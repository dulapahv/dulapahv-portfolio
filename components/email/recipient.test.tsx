import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { RecipientEmailTemplate } from './recipient';

describe('RecipientEmailTemplate', () => {
  const defaultProps = {
    name: 'John Doe',
    email: 'john@example.com',
    message: 'This is a test message'
  };

  it('should render the email template', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container).toBeInTheDocument();
  });

  it('should display the greeting', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('Hi DulapahV,');
  });

  it('should display notification message', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain(
      'You have a new message from the contact form on your portfolio website'
    );
  });

  it('should display sender full name', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('Full Name:');
    expect(container.textContent).toContain('John Doe');
  });

  it('should display sender email', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('Email:');
    expect(container.textContent).toContain('john@example.com');
  });

  it('should display the message content', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('Message:');
    expect(container.textContent).toContain('This is a test message');
  });

  it('should render multiline messages with line breaks', () => {
    const multilineMessage = 'Line 1\nLine 2\nLine 3';
    const { container } = render(
      <RecipientEmailTemplate
        name={defaultProps.name}
        email={defaultProps.email}
        message={multilineMessage}
      />
    );

    const html = container.innerHTML;
    expect(html).toContain('<br>');
  });

  it('should include mailto link for email', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    const html = container.innerHTML;
    expect(html).toContain('mailto:john@example.com');
  });

  it('should include portfolio link', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('DulapahV Portfolio');
  });

  it('should include GitHub link', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('GitHub');
  });

  it('should include LinkedIn link', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.textContent).toContain('LinkedIn');
  });

  it('should render with different email addresses', () => {
    const differentEmail = 'test@test.com';
    const { container } = render(
      <RecipientEmailTemplate
        name={defaultProps.name}
        email={differentEmail}
        message={defaultProps.message}
      />
    );
    expect(container.textContent).toContain(differentEmail);
  });

  it('should render with long name', () => {
    const longName = 'John Doe with a very long name that should still render correctly';
    const { container } = render(
      <RecipientEmailTemplate
        name={longName}
        email={defaultProps.email}
        message={defaultProps.message}
      />
    );
    expect(container.textContent).toContain(longName);
  });

  it('should render with long message', () => {
    const longMessage = 'This is a very long message '.repeat(50);
    const { container } = render(
      <RecipientEmailTemplate
        name={defaultProps.name}
        email={defaultProps.email}
        message={longMessage}
      />
    );
    expect(container.textContent).toContain(longMessage);
  });

  it('should handle special characters in message', () => {
    const messageWithSpecialChars = 'Hello! <>&"\'';
    const { container } = render(
      <RecipientEmailTemplate
        name={defaultProps.name}
        email={defaultProps.email}
        message={messageWithSpecialChars}
      />
    );
    expect(container.textContent).toContain('Hello!');
  });

  it('should render preview text with sender name', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container.innerHTML).toContain('New message from John Doe');
  });

  it('should match snapshot', () => {
    const { container } = render(<RecipientEmailTemplate {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
