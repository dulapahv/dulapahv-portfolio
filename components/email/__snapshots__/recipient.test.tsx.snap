// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecipientEmailTemplate > should match snapshot 1`] = `
<div>
  <div
    data-skip-in-text="true"
    style="display: none; overflow: hidden; line-height: 1px; opacity: 0; max-height: 0; max-width: 0;"
  >
    New message from John Doe
    <div>
       ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿
    </div>
  </div>
  <table
    align="center"
    border="0"
    cellpadding="0"
    cellspacing="0"
    role="presentation"
    style="margin: 2.5rem auto; max-width: 36rem; border-radius: 0.25rem; border-width: 1px; border-style: solid; border-color: rgb(229, 229, 229); padding: 1.25rem;"
    width="100%"
  >
    <tbody>
      <tr
        style="width: 100%;"
      >
        <td>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="margin-top: 1rem;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <img
                    alt="DulapahV Portfolio Logo"
                    height="48"
                    src="https://assets.dulapahv.dev/logo192.png"
                    style="display: block; outline: none; border: medium; text-decoration: none;"
                    width="48"
                  />
                  <hr
                    style="margin: 26px 0px; width: 100%; border: medium;"
                  />
                  <p
                    class="text-default-800"
                    style="text-align: left; font-size: 1rem; line-height: 1.5rem; margin-top: 16px; margin-bottom: 16px;"
                  >
                    <strong>
                      Hi DulapahV,
                    </strong>
                    <br />
                    You have a new message from the contact form on your portfolio website.
                    <br />
                    Please see the details below:
                  </p>
                  <p
                    style="font-size: 0.875rem; line-height: 1.5rem; color: rgb(0, 0, 0); margin-top: 16px; margin-bottom: 16px;"
                  >
                    <strong>
                      Full Name:
                    </strong>
                     
                    John Doe
                    <br />
                    <strong>
                      Email:
                    </strong>
                     
                    <a
                      href="mailto:john@example.com"
                      style="color: rgb(251, 86, 138); text-decoration-line: none;"
                      target="_blank"
                    >
                      john@example.com
                    </a>
                    <br />
                    <strong>
                      Message:
                    </strong>
                    <br />
                    <span>
                      This is a test message
                    </span>
                  </p>
                  <hr
                    style="margin: 26px 0px; width: 100%; border: medium;"
                  />
                  <p
                    style="font-size: 0.875rem; line-height: 1.5rem; color: rgb(0, 0, 0); margin-top: 16px; margin-bottom: 16px;"
                  >
                    <a
                      href="https://dulapahv.dev"
                      style="color: rgb(251, 86, 138); text-decoration-line: none;"
                      target="_blank"
                    >
                      DulapahV Portfolio
                    </a>
                     
                    ・
                     
                    <a
                      href="https://github.com/dulapahv"
                      style="color: rgb(251, 86, 138); text-decoration-line: none;"
                      target="_blank"
                    >
                      GitHub
                    </a>
                     
                    ・
                     
                    <a
                      href="https://www.linkedin.com/in/dulapahv"
                      style="color: rgb(251, 86, 138); text-decoration-line: none;"
                      target="_blank"
                    >
                      LinkedIn
                    </a>
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
